qiskit/scheduler/methods/basic.py

```diff
@@ -145,31 +147,46 @@ def translate_gates_to_pulse_defs(circuit: QuantumCircuit,
     circ_pulse_defs = []
 
     cmd_def = schedule_config.cmd_def
-    measured_qubits = set()  # Collect qubits that would like to be measured
+    m_slots = {}  # Map measured qubit index to clbit
 
     def get_measure_schedule() -> CircuitPulseDef:
         """Create a schedule to measure the qubits queued for measuring."""
         measures = set()
         all_qubits = set()
         sched = Schedule()
-        for q in measured_qubits:
-            measures.add(tuple(schedule_config.meas_map[q]))
+        for qubit in m_slots.keys():
+            measures.add(tuple(schedule_config.meas_map[qubit]))
         for qubits in measures:
+            unused_mem_slots = set(qubits) - set(m_slots.values())
```

Comment:
This is very much a nitpick, but can you move this down one line? `all_qubits.update` is a statement with a larger scope, so we should get it out of the way, and move `unused_mem_slots` a little bit closer to where it gets used.